src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1620,21 +1622,22 @@ public override bool Equals(object? comparand)
             // is passed, convert to Uri. This is inefficient, but allows us to
             // canonicalize the comparand, making comparison possible
             //
-            if ((object?)obj == null)
+            if (obj is null)
             {
-                string? s = comparand as string;
-
-                if ((object?)s == null)
+                if (!(comparand is string s))
                     return false;
 
+                if (ReferenceEquals(s, OriginalString))
+                    return true;
+
                 if (!TryCreate(s, UriKind.RelativeOrAbsolute, out obj))
                     return false;
             }
 
             // Since v1.0 two Uris are equal if everything but fragment and UserInfo does match
 
             // This check is for a case where we already fixed up the equal references
-            if ((object)_string == (object)obj._string)
+            if (ReferenceEquals(OriginalString, obj.OriginalString))
```

Comment:
Using `OriginalString` instead of `_string` makes this fast-path also work for Uris with non-ascii characters

Reply:
Is the comment above it still correct?  What does it mean to have "already fixed up the equal references"?

Reply:
I assume it's referring to the Uri instance being checked for reference equality with the comparand. I will remove it as I don't see it adding any value.